### DISCUSSION TITLE
Features don't save in QField

### DISCUSSION THREAD

**Discussion by jelleruben** [opengisch/QField#5055] [Q&A]
Hello,

I have a project with photolocations. I have made a Qfield package. and upload it to my Android phone.
I can open the package in Qfield, and I can add some new features to it. But how do I save it? 

If I upload the packege back to de pc (or I reset the QField app (fresh install)). The added features isn't there. So the new added features aren't saved in the package or shape. 

What I have done:

In Qgis (version 3.24.0) I go to the **Project properties** on the tab **Qfield**, follow by the tab **Cable Export**, I have set the photoloaction layer on "Offline editing" And the OSM layer stand on "Directly access data Source".

Then to make a package I go to  [Packeage Project for Qfield], I choice a folder by "Export Directory".

By "Advanced", I select the data folder (where I have my shape and images files) And click on [Create]

Then I upload the files folder to my Android phone.

I can open the project and can  add new features, but the features won't be save it in the shape.

I hope you understand my point, and hope you can help me. 

Thanks,

Jelle Ruben

**Comment by SeqLaz** [comment]
Hello @jelleruben, Is highly recommend that you transition your data source from shapefile (.shp) to GeoPackage (.gpkg). GeoPackage is the official data source format supported by QField, and it has many benefits such as performance in synchronization and fewer limitations.  
  
After finishing the survey in QField, you need to export the layers to a known location and then import them to QGIS through QFieldSync.  
  
For more information, please refer to the official documentation  

https://docs.qfield.org/get-started/storage/#exporting-modified-projects-and-datasets-on-android

**Comment by jelleruben** [comment]
@SeqLaz I have  have see on YouTube you can it do with shp files to. But I have now tried with a GeoPackage. And sadly in this option, my added points won't save it too.  I don't know what I'm doing wrong now. But these are the steps I do can you see (with screenshots) on https://jelleruben.nl/gis/qfield/

**Reply by SeqLaz** [reply]
Hello @jelleruben , upon reviewing the steps, everything appears to be correct. However, one crucial step that seems to be missing is exporting your modified layers or project to a location on the storage device. Once exported, you can then import these layers or projects onto your computer.

Additionally, it's worth mentioning that when you import a project, it creates a new copy of the project in the `<drive>:/Android/data/ch.opengis.qfield/` directory. Therefore, when you work on the imported project, you're not directly modifying the original project stored in `<drive>:/Android/Documents/`.

**Comment by jelleruben** [comment]
Hello @SeqLaz I am realy happy, you have looked at my steps, And I think I found it.
I didn't know the data was stored somewhere else.

So it works like this? I have my project on (for example)
\Samsung Note20 Ultra\SD-card\QField\Photolocation

If I added a feature of delete one, than find I my newe updated project on
\Samsung Note20 Ultra\Interne opslag\Android\data\ch.opengis.qfield\files\Imported Projects\QField

I don't think I read all the documentation on how this worked. Sorry.

But for you, realy thanks.

**Reply by SeqLaz** [reply]
It's great that your project is running, you can also connect with a USB cable and directly extract the information from
\Samsung Note20 Ultra\Interne opslag\Android\data\ch.opengis.qfield\files\Imported Projects\QField
